### PR TITLE
PTV-1913 force cookie resetting, including backup cookie

### PR DESCRIPTION
**Pull request by briehl**
# Description of PR purpose/changes

There's an issue where the `kbase_session_backup` cookie seems to be blocked or disappears on occasion. We're not entirely sure what's causing this right now - it could be some browser policy changes with regard to how the cookie is created. What it means is that direct HTTP communication to KBase services via iframes (i.e. report view and downloads) can fail by complaining that the user isn't logged in. This forces a new backup cookie to be created on page reload.

Unit and integration tests pending, but I wanted an image made to get in narrative-dev. That and production are the only environments that'll fail this way.

# Jira Ticket / Issue #

Related Jira ticket: https://kbase-jira.atlassian.net/browse/PTV-1913
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [ ] Tests pass locally and in GitHub Actions
- [ ] Changes available by spinning up a local narrative and navigating to _X_ to see _Y_

# Dev Checklist:

- [ ] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook
- [ ] (Python) I have run Ruff `format` and `check` on changed Python code manually or with a git precommit hook
- [ ] Any dependent changes have been merged and published in downstream modules

# Updating Version and Release Notes (if applicable)

- [ ] [Version has been bumped](https://semver.org/) for each release
- [ ] [Release notes](/RELEASE_NOTES.md) have been updated for each release (and during the merge of feature branches)

### CODE REVIEW

File: kbase-extension/static/kbase/js/api/auth.js [hunk 1]
```diff
@@ -277,6 +275,16 @@ define(['bluebird', 'jquery', 'narrativeConfig'], (Promise, $, Config) => {
...
+            // const options = {
+            //     method: callParams.method,
+            //     headers: {
+            //         Authorization: token,
+            //         'Content-Type': 'application/json',
+            //     },
+
+            // }
+            // return fetch(callString, options);
+
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/api/auth.js:287]: ??

Comment (briehl) [reply kbase-extension/static/kbase/js/api/auth.js:287]: The only vestige of jquery is that API call wrapped in a Promise, so I started replacing it with a `fetch` call. Then realized there's some downstream unpacking effects to manage. Then realized that was a silly exercise and wasn't solving anything useful so just put it back and stopped.

**Comment by briehl** [comment]
@ialarmedalien I think this is ready for a real review now.

**Review by dauglyon — COMMENTED** [review]
Looks good to me, but I'll leave it up to @ialarmedalien for final review as my narrative-familiarity is low. I am curious about the removal of 'use strict' as it can change error handling and variable declaration handling, but I assume it's safe...

**Comment by briehl** [comment]
@dauglyon the removal of 'use strict' comes from eslint. Compilers add it to builds, so it's not entirely necessary anymore in individual modules. ECMAScript modules are treated as strict by default: from https://eslint.org/docs/latest/rules/strict 

> In ECMAScript modules, which always have strict mode semantics, the directives are unnecessary.

The narrative codebase is on the old side, however, so now I'm thinking might not be entirely accurate? If that rule only applies to ECMAScript modules, and we're using AMDs still, is that just the eslint rule being finicky? Now I'm not sure!

**Comment by briehl** [comment]
It doesn't hurt anything being there, so I think I'll adjust the eslint rule to be safe.

I really can't wait to rewrite this stuff to be more modern...

**Review by ialarmedalien — APPROVED** [review]

**Pull request merged**